index.ts

```diff
@@ -0,0 +1,237 @@
+
+	export const arrayBuffer = isObjectOfType('ArrayBuffer');
+	export const sharedArrayBuffer = isObjectOfType('SharedArrayBuffer');
+
+	export const truthy = (x: any) => !!x; // eslint-disable-line no-implicit-coercion
+	export const falsy = (x: any) => !x;
+
+	// the tests are designed for Number.isNaN rather than global isNaN
+	export const nan = (x: any) => is.number(x) && isNaN(Number(x)); 
+	
+	export const nullOrUndefined = (x: any) => is.null_(x) || is.undefined(x);
+
+	const primitiveTypes = new Set<string>([
+		'undefined',
+		'string',
+		'number',
+		'boolean',
+		'symbol'
+	]);
+	export const primitive = (x: any) => is.null_(x) || primitiveTypes.has(typeof x);
+
+	export const integer = Number.isInteger;
+	export const safeInteger = Number.isSafeInteger;
+
+	export const plainObject = (x: any) => {
+		// From: https://github.com/sindresorhus/is-plain-obj/blob/master/index.js
+		let prototype;
+		// eslint-disable-next-line no-return-assign
+		return getObjectType(x) === 'Object' &&
+			(prototype = Object.getPrototypeOf(x), prototype === null ||
+				prototype === Object.getPrototypeOf({}));
+	};
+
+	export const iterable = (x: any) => !is.nullOrUndefined(x) && is.func(x[Symbol.iterator]);
+
+	export const class_ = (x: any) => is.func(x) && x.toString().startsWith('class ');
+
+	const typedArrayTypes = new Set<string>([
+		'Int8Array',
+		'Uint8Array',
+		'Uint8ClampedArray',
+		'Int16Array',
+		'Uint16Array',
+		'Int32Array',
+		'Uint32Array',
+		'Float32Array',
+		'Float64Array'
+	]);
+	export const typedArray = (x: any) => typedArrayTypes.has(getObjectType(x));
+
+	const isValidLength = (x: any) => is.safeInteger(x) && x > -1;
+	export const arrayLike = (x: any) => !is.nullOrUndefined(x) && !is.func(x) && isValidLength(x.length);
+
+	export const inRange = (x: number, range: number | number[]) => {
+		if (is.number(range)) {
+			return x >= Math.min(0, range as number) && x <= Math.max(range as number, 0);
+		}
+
+		if (is.array(range) && range.length === 2) {
+			// TODO: Use spread operator here when targeting Node.js 6 or higher
+			return x >= Math.min.apply(null, range) && x <= Math.max.apply(null, range);
+		}
+
+		throw new TypeError(`Invalid range: ${util.inspect(range)}`);
+	};
+
+	const NODE_TYPE_ELEMENT = 1;
+	const DOM_PROPERTIES_TO_CHECK = [
+		'innerHTML',
+		'ownerDocument',
+		'style',
+		'attributes',
+		'nodeValue'
+	];
+
+	export const domElement = (x: any) => is.object(x) && x.nodeType === NODE_TYPE_ELEMENT && is.string(x.nodeName) &&
+		!is.plainObject(x) && DOM_PROPERTIES_TO_CHECK.every(property => property in x);
+
+	export const infinite = (x: any) => x === Infinity || x === -Infinity;
+
+	const isAbsoluteMod2 = (value: number) => (x: number) => is.integer(x) && Math.abs(x % 2) === value;
+	export const even = isAbsoluteMod2(0);
+	export const odd = isAbsoluteMod2(1);
+
+	const isWhiteSpaceString = (x: any) => is.string(x) && /\S/.test(x) === false;
```

Comment:
Let's use a better name than `x`, since it will show up in auto-completion. We can use `value` instead.

Reply:
for all methods?

Reply:
Yes